### PR TITLE
Add docs for Name Resolution

### PR DESCRIPTION
**Pull request by Montoya**
Parts of the original PR have been cherry picked out to separate PRs that are already merged

This PR now only covers Name Resolution, which is still in refinement via the SIPs process. This PR is a draft until that is completed. 

Blocked by https://github.com/MetaMask/SIPs/pull/116

Fixes MetaMask/metamask-planning#1261

### CODE REVIEW

**Review by alexandratran — APPROVED** [review]
Made some minor stylistic edits, LGTM.

File: snaps/index.mdx
Comment (Montoya) [inline snaps/index.mdx:97]: Hi @alexandratran, we have a ton of cleanup to do on this but to be clear a Snap is always capitalized.

Comment (alexandratran) [reply snaps/index.mdx:97]: Okay, can we raise as a separate issue/PR to change this across the docs? Right now it is always lowercase when referencing a single snap.

Comment (Montoya) [reply snaps/index.mdx:97]: Yes that would be the way to solve it

**Comment by ziad-saab** [comment]
@SocketSecurity ignore memfs@3.6.0

**Review by ziad-saab — APPROVED** [review]

**Comment by Montoya** [comment]
OK I think this is ready to go now

**Comment by Montoya** [comment]
Everything looks great now, approved and ready to merge... but we cannot merge until the APIs land in Flask

**Comment by Montoya** [comment]
It looks like the only remaining change from this PR is for Name Resolution docs and then adding the feature cards on the home page for both name resolution and transaction severity @ziad-saab ?

**Review by alexandratran — APPROVED** [review]
Made some edits:
- Moved "Custom name resolution" to "Features"
- Added Flask label to this content and rearranged index page card
- Minor edits to content for style and consistency

LGTM

**Comment by ziad-saab** [comment]
@alexandratran my latest merge might have overwritten some of your minor edits.

File: snaps/reference/permissions.md
Comment (Montoya) [inline snaps/reference/permissions.md:163]: or **schemes such as** `farcaster:someuser`

Comment (Montoya) [reply snaps/reference/permissions.md:163]: I think mainnet is always lowercase? Can we double-check that?

Comment (ziad-saab) [reply snaps/reference/permissions.md:163]: image

on Ethereum.org it has a capital M. I'll fix scheme vs domain

Comment (Montoya) [reply snaps/reference/permissions.md]: OK!

Comment (alexandratran) [reply snaps/reference/permissions.md]: We use capitalized "Mainnet" throughout the docs (and other Consensys docs). Should leave this for now for consistency, but open to making it lowercase in the future.

**Comment by alexandratran** [comment]
@ziad-saab working on updating this again now

**Comment by alexandratran** [comment]
LGTM

**Review by Montoya — COMMENTED** [review]
LGTM

**Pull request merged**
